File: app/src/main/java/com/abstractclass/visitormanager/models/Person.java
package com.abstractclass.visitormanager.models;

import androidx.room.ColumnInfo;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Person implements Serializable {

    @ColumnInfo(name = "first_name")
    @SerializedName("first_name")
    private String firstName;

    @ColumnInfo(name = "middle_name")
    @SerializedName("middle_name")
    private String middleName;

    @ColumnInfo(name = "last_name")
    @SerializedName("last_name")
    private String lastName;

    @ColumnInfo(name = "national_id")
    @SerializedName("national_id")
    private String nationalId;

    @ColumnInfo(name = "passport_number")
    @SerializedName("passport_number")
    private String passportNumber;

    @ColumnInfo(name = "sex")
    @SerializedName("sex")
    private String sex;

    @ColumnInfo(name = "birth_date")
    @SerializedName("birth_date")
    private long birthDate;

    @ColumnInfo(name = "nationality")
    @SerializedName("nationality")
    private String nationality;

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getNationalId() {
        return nationalId;
    }

    public void setNationalId(String nationalId) {
        this.nationalId = nationalId;
    }

    public String getPassportNumber() {
        return passportNumber;
    }

    public void setPassportNumber(String passportNumber) {
        this.passportNumber = passportNumber;
    }

    public String getSex() {
        return sex;
    }

    public void setGender(int fifth_char) {
        switch(fifth_char) {
            case 0:
                this.setSex("female");
                break;
            case 1:
                this.setSex("male");
                break;
        }
    }

    public void setSex(String sex) {
        if(sex.toLowerCase().equals("m")) {
            this.sex = "male";
        } else if(sex.toLowerCase().equals("f")) {
            this.sex = "female";
        } else {
            this.sex = "undefined";
        }
    }

    public long getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(long birthDate) {
        this.birthDate = birthDate;
    }

    public String getNationality() {
        return nationality;
    }

    public void setNationality(String nationality) {
        this.nationality = nationality;
    }

    public String getMiddleName() {
        return middleName;
    }

    public void setMiddleName(String middleName) {
        this.middleName = middleName;
    }

    @Override
    public String toString() {
        return "Person{" +
                "firstName='" + firstName + '\'' +
                ", middleName='" + middleName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", nationalId='" + nationalId + '\'' +
                ", passportNumber='" + passportNumber + '\'' +
                ", sex='" + sex + '\'' +
                ", birthDate=" + birthDate +
                ", nationality='" + nationality + '\'' +
                '}';
    }
}

File: app/src/main/java/com/abstractclass/visitormanager/models/Person.kt
package com.abstractclass.visitormanager.models

import androidx.room.ColumnInfo
import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Person(
    @ColumnInfo(name = "first_name")
    @SerializedName("first_name")
    var firstName: String? = null,

    @ColumnInfo(name = "middle_name")
    @SerializedName("middle_name")
    var middleName: String? = null,

    @ColumnInfo(name = "last_name")
    @SerializedName("last_name")
    var lastName: String? = null,

    @ColumnInfo(name = "national_id")
    @SerializedName("national_id")
    var nationalId: String? = null,

    @ColumnInfo(name = "passport_number")
    @SerializedName("passport_number")
    var passportNumber: String? = null,

    @ColumnInfo(name = "sex")
    @SerializedName("sex")
    var sex: String? = null,

    @ColumnInfo(name = "birth_date")
    @SerializedName("birth_date")
    var birthDate: Long = 0,

    @ColumnInfo(name = "nationality")
    @SerializedName("nationality")
    var nationality: String? = null
) : Serializable

/**
class Person : Serializable {
    @ColumnInfo(name = "first_name")
    @SerializedName("first_name")
    private var firstName: String? = null
    @ColumnInfo(name = "middle_name")
    @SerializedName("middle_name")
    private var middleName: String? = null
    @ColumnInfo(name = "last_name")
    @SerializedName("last_name")
    private var lastName: String? = null
    @ColumnInfo(name = "national_id")
    @SerializedName("national_id")
    private var nationalId: String? = null
    @ColumnInfo(name = "passport_number")
    @SerializedName("passport_number")
    private var passportNumber: String? = null
    @ColumnInfo(name = "sex")
    @SerializedName("sex")
    private var sex: String? = null
    @ColumnInfo(name = "birth_date")
    @SerializedName("birth_date")
    private var birthDate: Long = 0
    @ColumnInfo(name = "nationality")
    @SerializedName("nationality")
    private var nationality: String? = null

    fun getFirstName(): String? {
        return firstName
    }

    fun setFirstName(firstName: String?) {
        this.firstName = firstName
    }

    fun getLastName(): String? {
        return lastName
    }

    fun setLastName(lastName: String?) {
        this.lastName = lastName
    }

    fun getNationalId(): String? {
        return nationalId
    }

    fun setNationalId(nationalId: String?) {
        this.nationalId = nationalId
    }

    fun getPassportNumber(): String? {
        return passportNumber
    }

    fun setPassportNumber(passportNumber: String?) {
        this.passportNumber = passportNumber
    }

    fun getSex(): String? {
        return sex
    }

    fun setGender(fifth_char: Int) {
        when (fifth_char) {
            0 -> setSex("female")
            1 -> setSex("male")
        }
    }

    fun setSex(sex: String?) {
        if (sex?.toLowerCase() == "m") {
            this.sex = "male"
        } else if (sex?.toLowerCase() == "f") {
            this.sex = "female"
        } else {
            this.sex = "undefined"
        }
    }

    fun getBirthDate(): Long {
        return birthDate
    }

    fun setBirthDate(birthDate: Long) {
        this.birthDate = birthDate
    }

    fun getNationality(): String? {
        return nationality
    }

    fun setNationality(nationality: String?) {
        this.nationality = nationality
    }

    fun getMiddleName(): String? {
        return middleName
    }

    fun setMiddleName(middleName: String?) {
        this.middleName = middleName
    }

    override fun toString(): String {
        return "Person{" +
                "firstName='" + firstName + '\'' +
                ", middleName='" + middleName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", nationalId='" + nationalId + '\'' +
                ", passportNumber='" + passportNumber + '\'' +
                ", sex='" + sex + '\'' +
                ", birthDate=" + birthDate +
                ", nationality='" + nationality + '\'' +
                '}'
    }
}
 */